src/nl/codefox/contextvh/knowledgebase.pl

```diff
@@ -52,31 +50,57 @@
+facilitiesID(StakeholderID) :- stakeholder(StakeholderID,'Voorzieningen',Money,Income).
+% stakeholder ID of duwo
+duwoID(StakeholderID) :- stakeholder(StakeholderID,'DUWO',Money,Income).
+% stakeholder ID of the tu
+tuID(StakeholderID) :- stakeholder(StakeholderID,'TU',Money,Income).
+
 % predicates to determine if land on the map is ours or others
 landOfOthers(MultiPoly) :- stakeholder(StakeholderID,'Private_Woningbouw_Burgers',_,_),
 	not(land(LandID,stakeholder(StakeholderID,_,_,_),MultiPoly,_,_)),
 	MultiPoly\=multipolygon('MULTIPOLYGON EMPTY').
 ourLand(MultiPoly) :- stakeholder(StakeholderID,'Private_Woningbouw_Burgers',_,_),
-	land(LandID,stakeholder(StakeholderID,_,_,_),MultiPoly,_,_),
+	(land(LandID,stakeholder(StakeholderID,_,_,_),MultiPoly,_,_);building(_,_,StakeholderID,_,_,_,_,MultiPoly,_,_)),
 	MultiPoly\=multipolygon('MULTIPOLYGON EMPTY').
 alreadyUpgraded(BuildingID) :- building(BuildingID,Name,_,_,_,_,_,_,_,_),sub_string(Name,_,3,_," + ").
 
+% zones
 zoneToRenovate(ZoneId):- needImprovement(3,ZoneId) ; (indicator(34,Value,Target,_),Value < Target).   
-
 zoneForGreens(ZoneId):- indicator(19,_,_,ZoneLinkList),member(zone_link(_,ZoneId,Current,Target),ZoneLinkList),Current<Target.
 
+% categories that are a member of housing
+housingBuildings(Categories) :- member('LUXE',Categories);member('NORMAL',Categories);member('SOCIAL',Categories);
+	member('OTHER',Categories);member('GARDEN',Categories);member('ROAD', Categories);member('INTERSECTION',Categories);
+	member('BRIDGE',Categories).
+% categories that are a member of categories that could be useful for the facilities party'
+facilitiesBuildings(Categories) :- member('SHOPPING',Categories); member('LEISURE',Categories); member('OFFICES',Categories).
+
+%%% Indicators
+ourIndicator(IndicatorID):- indicatorLink(2,Weights),member(indicatorWeights(IndicatorID,_,_),Weights).
 housingIndicator(IndicatorID):- ourID(OwnerID),indicatorLink(OwnerID,Weights),member(indicatorWeights(IndicatorID,IndicatorName,_),Weights),IndicatorName=="Building Privaat".
 
-% uses indicator scores to determine whether more Luxury houses are desirable,if this is not the case Normal houses are desirable.  
+% uses indicator scores to determine whether more Luxury houses are desirable,if this is not the case Normal houses are desirable.
 needLuxeHouse:- housingIndicator(ID),indicator(ID,_,_,ZoneLinkList),member(zone_link(0,_,Current1,Target1),ZoneLinkList),
 	indicator(ID,_,_,ZoneLinkList),member(zone_link(1,_,Current2,Target2),ZoneLinkList),
 	Target1-Current1>Target2-Current2.
-
+% indicator of the spatial quality
+qualityIndicator(Value,Target,ZoneID) :-
+	indicatorLink(_,IndicatorWeights), member(indicatorWeights(IndicatorID,IndicatorName,_),IndicatorWeights),
+	(IndicatorName=='Ruimtelijke kwaliteit'), indicator(IndicatorID,Value,Target,ZoneLink),
+	member(zone_link(ZoneID,IndicatorID,Value,Target),ZoneLink).
+% indicator of the sound
+soundIndicator(Value,Target,ZoneID) :-
```

Comment:
Although I think it is good to use predicates like this instead of hard-coding indicatorId's, I think this is a change that is outside of the scope of this PR. Please consider this in next PR's.